src/r_main.c

```diff
@@ -510,6 +510,11 @@ void R_ExecuteSetViewSize (void)
   // psprite scales
   pspritescale = FixedDiv(viewwidth_nonwide, SCREENWIDTH);       // killough 11/98
   pspriteiscale= FixedDiv(SCREENWIDTH, viewwidth_nonwide);       // killough 11/98
+  // [FG] make sure that the product of the weapon sprite scale factor
+  //      and its reciprocal is always at least FRACUNIT to
+  //      fix garbage lines at the top of weapon sprites
+  while (FixedMul(pspriteiscale, pspritescale) < FRACUNIT)
+    pspriteiscale++;
```

Comment:
How about just `pspriteiscale = FixedDiv(SCREENWIDTH, viewwidth_nonwide) + 1;`, like in `V_Init()`.

Reply:
Not strictly necessary for even dividers, but alright for the sake of simplicity and consistency. 👍